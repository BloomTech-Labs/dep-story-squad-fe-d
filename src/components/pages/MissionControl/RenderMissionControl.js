import React, { useEffect } from 'react';
import { Header } from '../../common';
import { Row, Col } from 'antd';
import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';

import draw_icon from '../../../assets/icons/draw_icon.svg';
import read_icon from '../../../assets/icons/read_icon.svg';
import write_icon from '../../../assets/icons/write_icon.svg';
import Checkbox from './Checkbox';

import { useOktaAuth } from '@okta/okta-react/dist/OktaContext';
import { getChildTasks, getStory } from '../../../api';

const RenderMissionControl = props => {
  const { push } = useHistory();
  const { authState } = useOktaAuth();

  useEffect(() => {
    if (props.tasks.id === null) {
      getChildTasks(authState, props.child.id, props.child.cohortId).then(
        res => {
          props.setTasks(res);
        }
      );
      getStory(authState, props.child.cohortId).then(res => {
        props.setSubmissionInformation(res);
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [authState]);

  // Will be for when we are checking whether or not the child has completed a task
  function handleChecked(e) {
    return `checked=${e.target.checked}`;
  }

  // redirects the user to the pdf of the story
  const handleReadStory = e => {
    e.stopPropagation();
    push('/child/story');
  };
  const handleWrite = e => {
    e.stopPropagation();
    push('/child/writing-sub');
  };
  const handleDraw = e => {
    e.stopPropagation();
    push('/child/drawing-sub');
  };

  return (
    <>
      <Header title="MISSION" />
      <div className="mission-container">
        <Row className="main-row">
          <Col className="read" xs={24} sm={12} onClick={handleReadStory}>
            <Checkbox
              className="checking-box"
              defaultChecked={false}
              onChange={handleChecked}
              isCompleted={props.tasks.hasRead}
            />

            <Col className="image-and-text-container">
              <img src={read_icon} alt="reading icon" />
              <p className="mission-control-text">Read</p>
            </Col>
          </Col>
          <Col className="write-and-draw" xs={24} sm={12}>
            <Row className="write" onClick={handleWrite}>
              <Checkbox
                className="checking-box"
                defaultChecked={false}
                onChange={handleChecked}
                isCompleted={props.tasks.hasWritten}
              />

              <Col className="image-and-text-container">
                <img
                  className="WritingandDrawingIcon"
                  src={write_icon}
                  alt="writing icon"
                />
                <p className="mission-control-text">Write</p>
              </Col>
            </Row>
            <Row className="draw" onClick={handleDraw}>
              <Checkbox
                className="checking-box"
                defaultChecked={false}
                onChange={handleChecked}
                isCompleted={props.tasks.hasDrawn}
              />
              <Col className="image-and-text-container">
                <img
                  className="WritingandDrawingIcon"
                  src={draw_icon}
                  alt="drawing icon"
                />
                <p className="mission-control-text">Draw</p>
              </Col>
            </Row>
          </Col>
        </Row>
      </div>
    </>
  );
};

export default connect(
  state => ({
    child: state.child,
    tasks: state.tasks,
  }),
  {}
)(RenderMissionControl);
