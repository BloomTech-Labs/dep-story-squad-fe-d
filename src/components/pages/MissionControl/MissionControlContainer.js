import React, { useState, useEffect, useMemo } from 'react';
import { useOktaAuth } from '@okta/okta-react';
import { connect } from 'react-redux';

import RenderMissionControl from './RenderMissionControl';
import { tasks } from '../../../state/actions';

const MissionControlContainer = ({ LoadingComponent, ...props }) => {
  console.log(props);
  const { authState, authService } = useOktaAuth();
  const [userInfo, setUserInfo] = useState(null);
  // eslint-disable-next-line
  const [memoAuthService] = useMemo(() => [authService], []);
  const [checked, setChecked] = useState(false);

  useEffect(() => {
    let isSubscribed = true;

    memoAuthService
      .getUser()
      .then(info => {
        // if user is authenticated we can use the authService to snag some user info.
        // isSubscribed is a boolean toggle that we're using to clean up our useEffect.
        if (isSubscribed) {
          setUserInfo(info);
        }
      })
      .catch(err => {
        isSubscribed = false;
        return setUserInfo(null);
      });
    return () => (isSubscribed = false);
  }, [memoAuthService]);

  const checkedToggle = e => {
    setChecked(!checked);
  };

  return (
    <>
      {authState.isAuthenticated && !userInfo && (
        <LoadingComponent message="Loading..." />
      )}
      {authState.isAuthenticated && userInfo && (
        <RenderMissionControl
          {...props}
          userInfo={userInfo}
          authService={authService}
          checkedToggle={checkedToggle}
        />
      )}
    </>
  );
};

// export default connect(
//   state => ({
//     child: state.child,
//     tasks: state.tasks,
//   }),
//   {}
// )(MissionControlContainer);

export default connect(null, {
  setTasks: tasks.setTasks,
})(MissionControlContainer);

// export default MissionControlContainer;
