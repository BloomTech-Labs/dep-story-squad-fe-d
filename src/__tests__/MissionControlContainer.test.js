import * as React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';

import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import LoadingComponent from '../components/common/ChildLoadingComponent';
import MissionControlContainer from '../components/pages/MissionControl/MissionControlContainer';
import RenderMissionControl from '../components/pages/MissionControl/RenderMissionControl';

jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => {
    return {
      authState: {
        isAuthenticated: true,
      },
      authService: {},
    };
  },
}));
configure({ adapter: new Adapter() });

describe('<MissionControlContainer />', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore();

  describe('Render <MissionControlContainer />', () => {
    let shallowWrapper = ShallowWrapper;
    beforeEach(() => {
      shallowWrapper = shallow(
        <Router>
          <Provider store={store}>
            <MissionControlContainer />
          </Provider>
        </Router>
      ).dive();
    });
    it('Find Render Mission', () => {
      expect(shallowWrapper.find(RenderMissionControl).length).toBe(0);
    });
    it('Find Loading Component', () => {
      expect(shallowWrapper.find(LoadingComponent));
    });
    it('MissionControlContainer', () => {
      expect(shallowWrapper).toMatchSnapshot(); // <------
    });
  });
});
