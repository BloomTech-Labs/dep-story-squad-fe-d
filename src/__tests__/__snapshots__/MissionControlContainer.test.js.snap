// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissionControlContainer /> Render <MissionControlContainer /> MissionControlContainer 1`] = `ShallowWrapper {}`;
